openedx/features/learner_profile/views/learner_achievements.py

```diff
@@ -34,9 +34,15 @@ def _get_ordered_certificates_for_user(self, request, username):
         Returns a user's certificates sorted by course name.
         """
         course_certificates = certificate_api.get_certificates_for_user(username)
+        passing_certificates = []
         for course_certificate in course_certificates:
-            course_key = course_certificate['course_key']
-            course_overview = get_course_overview_with_access(request.user, 'load', course_key)
-            course_certificate['course'] = course_overview
-        course_certificates.sort(key=lambda certificate: certificate['course'].display_name_with_default)
-        return course_certificates
+            if course_certificate.get('is_passing', False):
+                course_key = course_certificate['course_key']
+                try:
+                    course_overview = CourseOverview.get_from_id(course_key)
+                    course_certificate['course'] = course_overview
+                    passing_certificates.append(course_certificate)
+                except CourseOverview.DoesNotExist:
```

Comment:
Where does this belong?  Are certs not valid if the course is missing and should these be filtered by the call that returns the certs (optionally or otherwise)?  Or, are these valid certs and do we need to find some way to deliver them?  Just don't seem like the right place for this code.  This may affect where the test lives as well.